rust/shalom_dart_codegen/templates/operation.dart.jinja

```diff
@@ -181,5 +180,58 @@ class {{ root_type_name }}{
     }
 {% endfor %}
 
+// ------------ END OBJECT DEFINITIONS -------------
 
+class Request{{ root_selection }} extends Requestable {
+    final {{ root_selection }}Variables variables;
+
+    Request{{ root_selection }}({
+        required this.variables,
+    });
+
+    @override
+    Request toRequest() {
+        return Request(
+            query: r"""{{ operation.context.query }}""", 
+            variables: variables.toJson(), 
+            opType: OperationType.{{ operation.context.op_ty.name }}, 
+            StringopName: '{{ root_selection }}'
+        );
+    }
+}
+
+{% set variables = operation.context.variables %}
+class {{ root_selection }}Variables {
+    {% for name, variable in variables | items %}
+        final {{ type_name_for_variable(variable) }} {{ name }};
+    {% endfor %}
+
+    {{ root_selection }}Variables(
+        {% if variables | length > 0 %}
+            {
+            {% for name, variable in variables | items %}
+                {% if variable.is_optional %}
+                    this.{{name}} {% if variable.default_value is not none %} = Some(value: {{ variable.default_value}}) {% else %} = const None() {% endif %},
```

Comment:
as I said before there is not need to use Option<T> for optional variables that have defaults